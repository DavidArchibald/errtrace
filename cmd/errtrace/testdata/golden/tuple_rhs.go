package main

import "fmt"

func multipleValueErrAssignment() (err error) {
	defer func() {
		_, err = fmt.Println("Hello, World!")

		// Handles too few lhs variables
		err = fmt.Println("Hello, World!")

		// Handles too many lhs variables
		_, err, _ = fmt.Println("Hello, World!")

		// Handles misplaced err
		err, _ = fmt.Println("Hello, World!")
	}()

	return nil
}
