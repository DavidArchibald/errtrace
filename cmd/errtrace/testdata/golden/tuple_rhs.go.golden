package main

import "fmt"; import "braces.dev/errtrace"

func multipleValueErrAssignment() (err error) {
	defer func() {
		_, err = errtrace.Wrap2(fmt.Println("Hello, World!"))

		// Handles too few lhs variables
		err = errtrace.Wrap2(fmt.Println("Hello, World!"))

		// Handles too many lhs variables
		_, err, _ = errtrace.Wrap2(fmt.Println("Hello, World!"))

		// Handles misplaced err
		err, _ = errtrace.Wrap2(fmt.Println("Hello, World!"))
	}()

	return nil
}
